Source/Core/Core/CMakeLists.txt

```diff
@@ -218,12 +218,44 @@ add_library(core
   PowerPC/Interpreter/Interpreter_LoadStorePaired.cpp
   PowerPC/Interpreter/Interpreter_Paired.cpp
   PowerPC/Interpreter/Interpreter_SystemRegisters.cpp
-  PowerPC/Interpreter/Interpreter_Tables.cpp
   PowerPC/JitCommon/JitAsmCommon.cpp
   PowerPC/JitCommon/JitBase.cpp
   PowerPC/JitCommon/JitCache.cpp
 )
 
+add_executable(tablegen
+  PowerPC/TableGen.cpp
+)
+add_custom_command(
+  OUTPUT
+    Generated/OpID_Ranges.gen.h
+    Generated/OpID_DecodingTable.gen.cpp
+    Generated/OpInfo.gen.cpp
+    Generated/Interpreter_Table.gen.cpp
+    Generated/Cycles_Table.gen.cpp
+  COMMAND $<TARGET_FILE:tablegen> -i ${CMAKE_CURRENT_SOURCE_DIR}/../../../Data/instructions.csv
+    -ro Generated/OpID_Ranges.gen.h
+    -Do Generated/OpID_DecodingTable.gen.cpp
+    -No Generated/OpNames.gen.cpp
+    -0p OpType:: -o Generated/OpTypes.gen.cpp
+    -1o Generated/OpFlags.gen.cpp
+    -2d '*' -p Interpreter:: -o Generated/Interpreter_Table.gen.cpp
```

Comment:
I wonder if `Interpreter::` should be quoted or whether CMake is smart enough to not cause syntax errors down the road...